src/app/components/category-products/category-products.component.ts

```diff
@@ -1,10 +1,172 @@
-import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ViewEncapsulation,
+} from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { Observable, combineLatest, from, of } from 'rxjs';
+import {
+  filter,
+  map,
+  shareReplay,
+  startWith,
+  switchMap,
+  take,
+  tap,
+} from 'rxjs/operators';
+import { SortProductsOptionsQueryModel } from '../../query-models/sort-products-options.query-model';
+import { CategoryModel } from '../../models/category.model';
+import { ProductModel } from '../../models/product.model';
+import { CategoriesService } from '../../services/categories.service';
+import { ProductsService } from '../../services/products.service';
 
+interface PageParams {
```

Comment:
Create a separate file for this interface

Reply:
done